Source/WebCore/platform/graphics/cocoa/MediaPlaybackTargetContextCocoa.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if ENABLE(WIRELESS_PLAYBACK_TARGET)
+
+#include "MediaPlaybackTargetContext.h"
+#include <wtf/RetainPtr.h>
+
+OBJC_CLASS AVOutputContext;
+
+namespace WebCore {
+
+class MediaPlaybackTargetContextCocoa : public MediaPlaybackTargetContext {
+public:
+    WEBCORE_EXPORT MediaPlaybackTargetContextCocoa(RetainPtr<AVOutputContext>&&);
+    WEBCORE_EXPORT ~MediaPlaybackTargetContextCocoa();
+    WEBCORE_EXPORT MediaPlaybackTargetContextCocoa(const MediaPlaybackTargetContextCocoa&);
```

Comment:
We usually define operator= in case of copy constructor.

Reply:
Those operators shouldn’t be used. I had to define them and make them external as otherwise the IPC serialiser caused linkage error with missing vtable

Reply:
somehow it now compiles without them after I exported the base class default constructor/destructor.

Reply:
```4) Converting constructor. Constructs a variant holding the alternative type T_j that would be selected by overload resolution for the expression F([std::forward](http://en.cppreference.com/w/cpp/utility/forward)<T>(t)) if there was an overload of imaginary function F(T_i) for every T_i from Types... in scope at the same time, except that:
An overload F(T_i) is only considered if the declaration T_i x[] = { [std::forward](http://en.cppreference.com/w/cpp/utility/forward)<T>(t) }; is valid for some invented variable x.
```

https://godbolt.org/z/hW6v5oYxf

to be able to use a `MediaPlaybackTargetContextCocoa` in a std::variant, you can't have an explicit constructor.